url5997ccb760_JANNLab_JANNLab_tgz-pJ8-de_jannlab_examples_recurrent_SequentialXorExampleJ8/src/de/jannlab/generator/NetCoreGenerator.java

```diff
@@ -0,0 +1,1162 @@
+            final CellArray a = arrays[i];
+            //
+            if ((a.cellsnum <= 0) || (a.outdeg == 0)) continue;
+            final int albd = a.cellslbd;
+            final int aubd = a.cellsubd;
+            //
+            int lbd  = -1;
+            int ctr  = 0;
+            //
+            // find links range begin by src idx (swapped src and dst)
+            //
+            while (
+                    ((link + Link.IDX_DST) < this.structure.linksrev.length) &&
+                    (this.structure.linksrev[link + Link.IDX_DST] < albd)
+                    //((link + Link.IDX_SRC) < this.structure.linksrev.length) &&
+                    //(this.structure.linksrev[link + Link.IDX_SRC] < albd)
+            ) {
+                link += Link.LINK_SIZE;
+            }
+            lbd = link;
+            //
+            // find links range end by src idx (swapped src and dst).
+            //
+            while (
+                    ((link + Link.IDX_DST) < this.structure.linksrev.length) &&
+                    (this.structure.linksrev[link + Link.IDX_DST] <= aubd)
+                    //((link + Link.IDX_SRC) < this.structure.linksrev.length) &&
+                    //(this.structure.linksrev[link + Link.IDX_SRC] <= aubd)
+            ) {
+                link += Link.LINK_SIZE;
+                ctr++;
+            }
+            a.succslbd = lbd;
+            a.succsnum = ctr;
+            //
+            // changed: 01.02.2012, Sebastian Otte
+            // a.succsubd = (a.succslbd + a.succsnum) - 1;
+            //
+            a.succsubd = link - Link.LINK_SIZE;
+        }
+        this.structure.arrays    = arrays;
+        this.structure.arraysnum = arrays.length;
+    }
+    
+    /**
+     * Setup up cell to X maps.
+     */
+    private void setupMaps() {
+        //
+        this.layermap = new int[this.cellsoffset];
+        this.arraymap = new int[this.cellsoffset];
+        this.indeg    = new int[this.cellsoffset];
+        this.outdeg   = new int[this.cellsoffset];
+        //
+        //
+        for (int i = 0; i < this.cellsoffset; i++) {
+            this.layermap[i] = NOT_ASSIGNED;
+            this.arraymap[i] = NOT_ASSIGNED;
+        }
+        //
+        // determine layer assignment.
+        //
+        for (int i = 0; i < this.layers.size(); i++) {
+            final Layer layer = this.layers.get(i);
+            //
+            for (int j = layer.cellslbd; j <= layer.cellsubd; j++) {
+                this.layermap[j] = i;
+            }
+        }
+        //
+        // determine array assignment.
+        //
+        for (int i = 0; i < this.arrays.size(); i++) {
+            final CellArray array = this.arrays.get(i);
+            //
+            for (int j = array.cellslbd; j <= array.cellsubd; j++) {
+                this.arraymap[j] = i;
+            }
+        }
+    }
+    
+    /**
+     * Setup layers. This is mainly the computation order.
+     */
+    private void setupLayers() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 459-522 into a separate method.